base_address_parish/models/res_partner.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 Riverminds Cia Ltda - Mamfredy Mejia Matute
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from lxml import etree
+
+from odoo import api, models, fields
+from odoo.tools.translate import _
+
+
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    country_enforce_parishes = fields.Boolean(related='country_id.enforce_parishes', readonly=True)
+    parish_id = fields.Many2one('res.parish', string='Parish')
+
+    @api.onchange('parish_id')
+    def _onchange_parish_id(self):
+        if self.parish_id:
+            self.zip = self.parish_id.code
```

Comment:
If code of parish is the zip on partner, field should has same name, I'm not sure if this implementation is rigth because zip code can be different on the same parish, please share us the base for this design

Reply:
Zip code is different, you can check this here: https://www.codigopostal.gob.ec/

I think they should not be related to the parish code.